bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/servlets/GraphqlClientDataSourceServlet.java

```diff
@@ -0,0 +1,137 @@
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+package com.adobe.cq.commerce.core.components.internal.servlets;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import javax.servlet.Servlet;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.SyntheticResource;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.api.wrappers.ValueMapDecorator;
+import org.jetbrains.annotations.NotNull;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.adobe.granite.ui.components.ds.DataSource;
+import com.adobe.granite.ui.components.ds.SimpleDataSource;
+import com.day.cq.i18n.I18n;
+
+@Component(
+    service = { Servlet.class },
+    property = {
+        "sling.servlet.resourceTypes=" + GraphqlClientDataSourceServlet.RESOURCE_TYPE,
+        "sling.servlet.methods=GET",
+        "sling.servlet.extensions=html"
+    })
+public class GraphqlClientDataSourceServlet extends SlingSafeMethodsServlet {
+
+    public final static String RESOURCE_TYPE = "core/cif/components/page/v1/datasource/graphqlclients";
+
+    @Reference
+    ConfigurationAdmin configurationAdmin;
+
+    private I18n i18n;
+
+    @Override
+    protected void doGet(@NotNull SlingHttpServletRequest request, @NotNull SlingHttpServletResponse response) {
+        i18n = new I18n(request);
+        SimpleDataSource graphqlClientDataSource = new SimpleDataSource(getGraphqlClients(request).iterator());
+        request.setAttribute(DataSource.class.getName(), graphqlClientDataSource);
+    }
+
+    protected List<Resource> getGraphqlClients(@NotNull SlingHttpServletRequest request) {
+        ResourceResolver resolver = request.getResourceResolver();
+        List<Resource> graphqlClients = new ArrayList<>();
+
+        List<Configuration> configs = null;
+        try {
+            configs = Arrays.asList(
+                configurationAdmin.listConfigurations("(service.factoryPid=com.adobe.cq.commerce.graphql.client.impl.GraphqlClientImpl)"));
```

Comment:
When possible, I think we should avoid hard-coding implementation specific references. You could replace this with something similar to https://github.com/adobe/commerce-cif-connector/blob/master/bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/graphql/magento/GraphqlDataServiceImpl.java#L76-L94 where you can only keep a `Set` of identifiers.